src/Encryption.php

```diff
@@ -318,4 +305,96 @@ private static function getIKM(string $userAuthToken, string $userPublicKey, str
 
         return $sharedSecret;
     }
+
+    /**
+     * @param string $userPublicKey
+     * @param string $localPublicKey
+     * @param PrivateKey $privateKey
+     *
+     * @return string
+     */
+    private static function computeSharedSecret(string $userPublicKey, string $localPublicKey, PrivateKey $privateKey): string
+    {
+        $curve = NistCurve::curve256();
+        if (\function_exists('openssl_pkey_derive')) {
+            try {
+                $publicPem = self::publicKeyToPem($userPublicKey);
+                $privatePem = self::privateKeyToPem($localPublicKey, $privateKey);
```

Comment:
This still uses a gmp function as is

Reply:
Yes, but here we convert a JWK to a PEM.
In the other parts we create these JSON Web keys. This is something that could be removed in order to deal only with PEM keys generated by OpenSSL.
The hidden goal here is to get rid on the web-token components and perform cypher operations with native PHP functions or OpenSSL extension. This will limit the number of dependencies and increase the global performance of the library.
Unfortunatly, this cannot be done blindly and tests shall pass first.